contrib/examples/kafka/src/main/java/com/google/cloud/dataflow/contrib/kafka/examples/TopHashtagsExample.java

```diff
@@ -0,0 +1,257 @@
+    @Description("Sliding window length in minutes")
+    @Default.Integer(10)
+    Integer getSlidingWindowLengthMinutes();
+    void setSlidingWindowLengthMinutes(Integer value);
+
+    @Description("Trigger window interval in minutes")
+    @Default.Integer(1)
+    Integer getSlidingWindowIntervalMinutes();
+    void setSlidingWindowIntervalMinutes(Integer value);
+
+    @Description("Bootstrap Server(s) for Kafka")
+    @Required
+    String getBootstrapServers();
+    void setBootstrapServers(String servers);
+
+    @Description("One or more comma seperated topics to read from")
+    @Required
+    List<String> getTopics();
+    void setTopics(List<String> topics);
+
+    @Description("Number of Top Hashtags to track")
+    @Default.Integer(10)
+    Integer getNumTopHashtags();
+    void setNumTopHashtags(Integer count);
+
+    @Description("Kafka topic name for writing results")
+    @Required
+    String getOutputTopic();
+    void setOutputTopic(String topic);
+  }
+
+  public static void main(String args[]) {
+
+    Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+
+    Pipeline pipeline = Pipeline.create(options);
+
+    KafkaIO.Read<?, String> reader = KafkaIO.read()
+        .withBootstrapServers(options.getBootstrapServers())
+        .withTopics(options.getTopics())
+        .withValueCoder(StringUtf8Coder.of())
+        .withTimestampFn(timestampFn);
+
+    pipeline
+      .apply("sample_tweets", reader)
+      .apply(Values.<String>create())
+      .apply(ParDo.of(new ExtractHashtagsFn()))
+      .apply(Window.<String>into(SlidingWindows
+          .of(Duration.standardMinutes(options.getSlidingWindowLengthMinutes()))
+          .every(Duration.standardMinutes(options.getSlidingWindowIntervalMinutes()))))
+      .apply(Count.<String>perElement())
+      .apply(Top.of(options.getNumTopHashtags(), new KV.OrderByValue<String, Long>())
+                .withoutDefaults())
+      .apply(ParDo.of(new OutputFormatter()))
+      .apply(ParDo.of(new KafkaWriter(options)));
+
+    pipeline.run();
+  }
+
+  // The rest of the file implements DoFns to extract hashtags, formatting output, writing output
+  // back to Kafka. Note that writing to Kafka is not a complete Dataflow Sink. It is a best-effort
+  // logging of the results.
+
+  private static final ObjectMapper jsonMapper = new ObjectMapper();
+
+  /**
+   * Emit hashtags in the tweet (if any).
+   */
+  private static class ExtractHashtagsFn extends DoFn<String, String> {
+
+    @Override
+    public void processElement(ProcessContext ctx) throws Exception {
+      for (JsonNode hashtag : jsonMapper.readTree(ctx.element())
+          .with("entities")
+          .withArray("hashtags")) {
+        ctx.output(hashtag.get("text").asText());
+      }
+    }
+  }
+
+  // extract timestamp from "timestamp_ms" field.
+  private static SerializableFunction<KV<byte[], String>, Instant> timestampFn =
```

Comment:
`final`?

Maybe name it something like `TWEET_TIMESTAMP_OR_NOW`?

Maybe Javadoc it to explain what it does (timestamp or now)?